app/javascript/shopify_app/turbolinks.js

```diff
@@ -0,0 +1,58 @@
+import { getSessionToken } from "@shopify/app-bridge-utils";
+
+const SESSION_TOKEN_REFRESH_INTERVAL = 2000; // Request a new token every 2s
+
+document.addEventListener("turbolinks:request-start", function (event) {
+  var xhr = event.data.xhr;
+  xhr.setRequestHeader("Authorization", "Bearer " + window.sessionToken);
+});
+
+document.addEventListener("turbolinks:render", function () {
+  $("form, a[data-method=delete]").on("ajax:beforeSend", function (event) {
```

Comment:
I believe this won't really happen with turbolinks apps, but would it be a problem if apps dynamically add elements that might require authentication?

I'm mainly thinking about the scenario where an app creates a modal with a form in it - will it be up to the app to make sure it's passing the header along in that case?

Reply:
I think the idea with turbolinks is that all form submissions need to be handled by js. They can't just be straight browser-native form submissions.